packages/web/src/components/segmented-progress-bar/SegmentedProgressBar.tsx

```diff
@@ -0,0 +1,83 @@
+import { Box, Flex, useTheme } from '@audius/harmony'
+import { range } from 'lodash'
+
+interface SegmentedProgressBarProps {
+  numSteps: number
+  stepsComplete: number
+  isCompact?: boolean
+}
+
+/**
+ * `SegmentedProgressBar` displays a configurable amount of progress segments
+ * indicating progress through some task.
+ */
+export const SegmentedProgressBar = ({
+  numSteps,
+  stepsComplete,
+  isCompact = false
+}: SegmentedProgressBarProps) => {
+  const { color, spacing } = useTheme()
+  /**  Div hierarchy explanation:
+   *
+   * - Outermost div .container provides the grey border
+   *
+   * - Middle div .segmentContainerMask is inset from .container, providing the white 'gap' between the grey border
+   *   and the segments, as well as masking the overflowing child .segment_conatiner div with `overflow: hidden`
+   *
+   * - Innermost div .segmentContainer contains the segments (wow!) and is slightly wider than the .segmentContainerMask
+   *  so that the segments can 'fill out' the rounded corners of the .segmentContainerMask
+   */
+  return (
+    <Flex
+      borderRadius={isCompact ? 'm' : 'l'}
+      border='strong'
+      backgroundColor='white'
+      css={{
+        borderWidth: spacing.unitHalf,
+        position: 'relative',
+        overflow: 'hidden'
+      }}
+    >
+      <Flex
+        w={isCompact ? 144 : 206}
+        h={isCompact ? spacing.unit6 : spacing.unit9}
```

Comment:
nit: this is reused below, can make a var

Reply:
cools